testsuite/MDAnalysisTests/analysis/test_psa.py

```diff
@@ -292,10 +292,41 @@ def test_distances_from_hausdorff_pairs_frames(self, psa):
                             dists[self.iu1],
                             decimal=6, err_msg=err_msg)
 
+
 class TestPSAExceptions(object):
     '''Tests for exceptions that should be raised
     or caught by code in the psa module.'''
 
+    @pytest.fixture()
+    def universe_no_path(self):
+        universe1 = mda.Universe(PSF, DCD)
+        return universe1
+
+    def test_no_path_data(self, universe_no_path, tmpdir):
+        """Test that ValueError is raised when no path data. Covers
+        get_num_atoms() in class Path and get_num_atoms(),
+        get_num_paths(), and get_paths() in class PSAnalysis"""
+        match_exp = "No path data"
+        path = PSA.Path(universe_no_path, universe_no_path)
+        psa = PSA.PSAnalysis([universe_no_path, universe_no_path,
+                              universe_no_path],
+                             path_select='name CA',
+                             targetdir=str(tmpdir))
+
+        # Case where user did not run Path.to_path()
+        with pytest.raises(ValueError, match=match_exp):
+            path.get_num_atoms()
+
+        # Cases where user did not run PSAnalysis.generate_paths()
+        with pytest.raises(ValueError, match=match_exp):
+            psa.get_num_atoms()
+
```

Comment:
Can you add a quick comment for last two cases?

Reply:
For sure! I just finished adding new comments. I also tried to make the comments more clear.